sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -2570,7 +2570,7 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
 
     assert(e3.getCause.isInstanceOf[RuntimeException])
     assert(e3.getCause.getMessage.matches(
-      "\\[USER_RAISED_EXCEPTION\\] '\\(a#\\d+ > b#\\d+\\)' is not true!"))
+      "\\[USER_RAISED_EXCEPTION\\] '\\(a#\\d+ > b#\\d+\\)' is not true! SQLSTATE: P0001"))
```

Comment:
Could you try to use a regexp to match the SQLSTATE?

Reply:
Just wonder how such regexp should look like. Maybe put it to PR's description.

Reply:
Ugh. I'll try

Reply:
I'm thinking that I want to keep it. SQLSTATE is not really supposed to change. And in this case, we expect people to catch them since it's a user-raised exception.